miio/g1vacuum.py

```diff
@@ -0,0 +1,312 @@
+                self.clean_area,
+                self.clean_time,
+                self.total_clean_count,
+            )
+        )
+        return ret
+
+class G1Vacuum(MiotDevice):
+    """Support for G1 vacuum (G1, mijia.vacuum.v2)."""
+
+    def __init__(
+            self,
+            ip: str = None,
+            token: str = None,
+            start_id: int = 0,
+            debug: int = 0,
+            lazy_discover: bool = True,
+    ) -> None:
+        super().__init__(_MAPPING, ip, token, start_id, debug, lazy_discover)
+
+    @command(
+        default_output=format_output(
+            "",
+            "State: {result.state}\n"
+            "Error: {result.error}\n"
+            "Battery: {result.battery}%\n"
+            "Mode: {result.operating_mode}\n"
+            "Mop State: {result.mop_state}\n"
+            "Charge Status: {result.charge_state}\n"
+            "Fan speed: {result.fan_speed}\n"
+            "Water level: {result.water_level}\n"
+            "Filter Life Level: {result.filter_life_level}%\n"
+            "Brush Life Level: {result.brush_life_level}%\n"
+            "Side Brush Life Level: {result.brush_life_level2}%\n"
+            "Clean Area: {result.clean_area}\n"
+            "Clean Time: {result.clean_time}\n"
+            #  "Total Clean Area: {result.total_clean_area}\n"
+            "Total Clean Count: {result.total_clean_count}\n",
+        )
+    )
+
+    def status(self) -> G1Status:
+        """Retrieve properties."""
+
+        return G1Status(
+            {
+                prop["did"]: prop["value"] if prop["code"] == 0 else None
+                for prop in self.get_properties_for_mapping()
+            }
+        )
+
+    def call_action(self, siid, aiid, params=None):
+        """Call Action"""
+        # {"did":"<mydeviceID>","siid":18,"aiid":1,"in":[{"piid":1,"value":2}]
+        if params is None:
+            params = []
+        payload = {
+            "did": f"call-{siid}-{aiid}",
+            "siid": siid,
+            "aiid": aiid,
+            "in": params,
+        }
+        return self.send("action", payload)
+
+    @command()
+    def return_home(self) -> None:
+        """Return Home."""
+        return self.call_action(2, 3)
+
+    @command()
+    def start(self) -> None:
+        """Start Cleaning"""
+        return self.call_action(2, 1)
+
+    @command()
+    def stop(self) -> None:
+        """Stop Cleaning"""
+        return self.call_action(2, 2)
+
+    @command()
+    def find(self) -> None:
+        """Find the robot."""
+        return self.call_action(6, 1)
+
+    @command()
+    def reset_brush_life(self) -> None:
+        """Reset Brush Life."""
+        return self.call_action(14, 1)
+
+    @command()
+    def reset_filter_life(self) -> None:
+        """Reset Filter Life."""
+        return self.call_action(11, 1)
+
+    @command()
+    def reset_brush_life2(self) -> None:
+        """Reset Brush Life"""
+        return self.call_action(15, 1)
```

Comment:
It would be great if you could replace these with a single method following the same API as rockrobo/vacuum.py uses: https://github.com/rytilahti/python-miio/blob/master/miio/vacuum.py#L303

Reply:
I somehow converted it to a single method, but I am not quite sure that my hardwired conditonal construct is any good.